lib/Bio/EnsEMBL/DataCheck/Checks/SchemaVersion.pm

```diff
@@ -0,0 +1,57 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::SchemaVersion;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME        => 'SchemaVersion',
+  DESCRIPTION => 'make sure the schema version is the same as the DB name',
+  GROUPS      => ['EGCoreHandover','EGVariationHandover', 'EGOtherFeaturesHandover', 'EGFuncgen'],
+  DB_TYPES    => ['core', 'variation', 'funcgen', 'otherfeatures']
```

Comment:
This could also apply to compara dbs, and rnaseq and cdna dbs. For truly universal datachecks, you can omit this parameter, in which case the test is assumed to be applicable to all types of database.